File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
        def removeElement(self, nums: List[int], val: int) -> int:
                left = 0
                right = len(nums) - 1

                while left <= right:
                    while left <= right and nums[left] != val:
                        left = left + 1

                    while left <= right and nums[right] == val:
                        right = right - 1

                    if left < right:
                        nums[left] = nums[right]
                        nums[right] = val
                        left = left + 1
                        right = right - 1
                return left

File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
private:
    void dfs(int i, vector<int>& nums, vector<vector<int>>& res, vector<int> &subset){
        if (i >= nums.size()){
            res.push_back(subset);
            return;
        }

        subset.push_back(nums[i]);
        dfs(i + 1, nums, res, subset);

        subset.pop_back();
        dfs(i + 1, nums, res, subset);
    }

public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> subset;
        dfs(0,nums,res,subset);
        return res;
    }
};